docs/io/fits/usage/cloud.rst

```diff
@@ -0,0 +1,268 @@
+it.  This works fine for small files but tends to require a lot of time and
+memory for large files.
+
+You can improve the performance for large files by passing the parameters
+``use_fsspec=True`` and ``lazy_load_hdus=True`` to `open`.  This will make
+Astropy use ``fsspec`` to download only the necessary parts of the FITS file.
+For example:
+
+.. doctest-requires:: fsspec
+
+    >>> from astropy.io import fits
+    ...
+    >>> # `fits.open` will download the primary header
+    >>> with fits.open(url, use_fsspec=True, lazy_load_hdus=True) as hdul:  # doctest: +REMOTE_DATA
+    ...
+    ...     # Download a single header
+    ...     header = hdul[1].header
+    ...
+    ...     # Download a single data array
+    ...     image = hdul[1].data
+    ...
+    ...     # Download a 10-by-20 pixel cutout by using .section
+    ...     cutout = hdul[2].section[10:20, 30:50]
+
+The example above requires less time and memory than would be required to
+download the entire file. This is because ``fsspec`` is able to leverage
+two *lazy data loading* features available in Astropy:
+
+1. The ``lazy_load_hdus`` parameter takes care of loading HDU header and data
+   attributes on demand rather than reading all HDUs at once.  This parameter
+   is set to ``True`` by default. You do not need to pass it explicitely,
+   unless you changed its default value in the :ref:`astropy:astropy_config`.
+2. The `ImageHDU.section` property enables a subset of a data array to be
+   read into memory without downloading the entire image or cube. See the
+   :ref:`astropy:data-sections` part of the documentation for more details.
+
+Additional tips for achieving good performance when working with remote files
+are provided in the :ref:`astropy:optimizing_fsspec` section further down
+this page.
+
+.. note::
+
+    The `ImageHDU.section` feature is only available for uncompressed FITS
+    image extensions.  This includes file-level compression like gzip as
+    well as compression internal to the FITS format, like tile compression.
+    Attempting to use ``.section`` on a compressed image will yield an
+    `AttributeError`.
+
+
+Subsetting FITS files hosted in Amazon S3 cloud storage
+=======================================================
+
+The FITS file used in the example above also happens to be available via
+Amazon cloud storage, where it is stored in a
+`public S3 bucket <https://registry.opendata.aws/hst/>`__ at the following
+location::
+
+    >>> s3_uri = "s3://stpubdata/hst/public/j8pu/j8pu0y010/j8pu0y010_drc.fits"
+
+With ``use_fsspec`` enabled, you can obtain a small cutout from a file stored
+in Amazon S3 cloud storage in the same way as above.  For example:
+
+.. doctest-requires:: fsspec
+
+    >>> # Download a small 10-by-20 pixel cutout from a FITS file stored in Amazon S3
+    >>> with fits.open(s3_uri, use_fsspec=True, fsspec_kwargs={"anon": True}) as hdul:  # doctest: +REMOTE_DATA
```

Comment:
Again, I'd not bother adding arguments that are set by default.

Reply:
Agreed.  I will go ahead and modify the docs accordingly.

Reply:
Fixed via 7e7d668852e800d5fce640c020309ccbc7ea4607.